### PR TITLE
document AlwaysPullImages admission plugin as protection against imagePullPolicy: Never

### PR DESCRIPTION
**Pull request by mburke5678**
https://bugzilla.redhat.com/show_bug.cgi?id=1621911

### CODE REVIEW

**Comment by mburke5678** [comment]
@sjenning

I added a section to the "Configuring Global Build Defaults and Overrides" section, as Ben alluded to in [BZ comment 3](https://bugzilla.redhat.com/show_bug.cgi?id=1621911#c3). Not sure if this is where the information should land.

Also added a note and a link to the Managing Images topic where we discuss `imagePullPolicy`. 

Please let me know what you think.

**Comment by mburke5678** [comment]
@anpingli PTAL

**Comment by anpingli** [comment]
@xingxingxia

**Comment by xingxingxia** [comment]
Per https://url.corp.redhat.com/222da39 (RH internal), this plugin is owned by dev-exp, @wzheng1 , could you help check this PR? Thanks.

**Comment by mburke5678** [comment]
Verified in BZ https://bugzilla.redhat.com/show_bug.cgi?id=1621911#c13
I will create a separate PR to change the restart command in 3.9, https://github.com/openshift/openshift-docs/pull/17289#issuecomment-545802362

@openshift/team-documentation   PTAL

File: admin_guide/image_policy.adoc [hunk 1]
```diff
@@ -233,6 +233,36 @@ endif::[]
 value in your `master-config.yaml` file.
 ====
 
+[[image-policy-always-pull-images]]
+== Using an Admission Controller to Always Pull Images
+
+After an image is pulled to a node, any Pod on that node from any user can use the image without an authorization check against the image. 
+To ensure that Pods do not use images for which they do not have credentials, use the *AlwaysPullImages* admission controller. 
+
+This xref:../architecture/additional_concepts/admission_controllers.adoc#architecture-additional-concepts-admission-controllers[admission controller] modifies every new Pod to force the image pull policy to `Always`, ensuring that private images can only be used by those who have the credentials to pull them, even if the Pod specification uses a xref:../dev_guide/managing_images.adoc#image-pull-policy[image pull policy] of `Never`.
```
Comment (bobfuru) [inline admin_guide/image_policy.adoc]: s/uses a/uses an/

File: admin_guide/image_policy.adoc [hunk 2]
```diff
@@ -233,6 +233,36 @@ endif::[]
...
+admissionConfig:
+  pluginConfig:
+    AlwaysPullImages: <1> 
+      configuration:
+        kind: DefaultAdmissionConfig
+        apiVersion: v1
+        disable: false <2>
+----
+<1> Admission plug-in name.
+<2> Specify `false` to indicate the plug-in should be enabled.
```
Comment (bobfuru) [inline admin_guide/image_policy.adoc]: s/indicate the/indicate that the/

File: admin_guide/image_policy.adoc [hunk 3]
```diff
@@ -233,6 +233,36 @@ endif::[]
...
+
+. Restart master services running in control plane static pods using the `master-restart` command:
```
Comment (bobfuru) [inline admin_guide/image_policy.adoc]: s/pods/Pods/

File: dev_guide/managing_images.adoc [hunk 4]
```diff
@@ -490,6 +490,13 @@ parameter is not specified, {product-title} sets it based on the image's tag:
 . Otherwise, {product-title} defaults `*imagePullPolicy*` to `*IfNotPresent*`.
 endif::[]
 
+[NOTE]
+====
+When using the `*Never*` Image Pull Policy, you can ensure that private images can only be used by pods with credentials to pull those images
```
Comment (bobfuru) [inline dev_guide/managing_images.adoc]: s/pods/Pods/

File: dev_guide/managing_images.adoc [hunk 5]
```diff
@@ -490,6 +490,13 @@ parameter is not specified, {product-title} sets it based on the image's tag:
...
+using the xref:../admin_guide/image_policy.adoc#image-policy-always-pull-images[*AlwaysPullImages* admission controller].
+If this admission controller is not enabled, any pod from any user on a node can use the image without any authorization check against the image.
```
Comment (bobfuru) [inline dev_guide/managing_images.adoc]: s/pod/Pod/

**Review by bobfuru — COMMENTED** [review]
A few small edits.

File: dev_guide/managing_images.adoc (hunk 4 above)
Comment (mburke5678) [reply dev_guide/managing_images.adoc]: @bobfuru I am going to leave these as lower-case. The other references to _pod_ are all lower-case in this topic.

File: dev_guide/managing_images.adoc (hunk 5 above)
Comment (mburke5678) [reply dev_guide/managing_images.adoc]: @bobfuru I am going to leave these as lower-case. The other references to _pod_ are all lower-case in this topic.

**Pull request merged**